### ISSUE TITLE
Codecov integration doesn't work

### ISSUE THREAD

**Issue by outofforest** [Reviewable/Reviewable#1081]
Hello team!

I tried to integrate Reviewable with Codecov today and seems it doesn't work.
Report is uploaded to Codecov and presented there but it is not fetched by Reviewable.
I think the reason is that you use deprecated v1 API instead of v2: https://about.codecov.io/blog/getting-started-with-the-codecov-api-v2/

**Comment by pkaminski** [comment]
Thanks for reporting this!  In principle it should be easy for Reviewable to adapt to this new API but I'm having trouble retrieving per-line coverage info from Codecov.  Here's what I tried:

[This sample PR](https://github.com/codecov/example-typescript/pull/14) shows a [report](https://app.codecov.io/gh/codecov/example-typescript/pull/14) with per-line coverage info.  However, when I try to use the [commit coverage report API](https://docs.codecov.com/reference/repos_report_retrieve) (which explicitly claims to return line-by-line coverage info), I only get the following:

```
$ curl 'https://api.codecov.io/api/v2/github/codecov/repos/example-typescript/totals/?sha=69441683464ac4ba859654416c77fa0d4e732882' -H 'accept: application/json'

{
    "totals": {
        "files": 1,
        "lines": 10,
        "hits": 8,
        "misses": 1,
        "partials": 1,
        "coverage": 80,
        "branches": 1,
        "methods": 0,
        "messages": 0,
        "sessions": 3,
        "complexity": 0,
        "complexity_total": 0,
        "complexity_ratio": 0,
        "diff": [
            0,
            0,
            0,
            0,
            0,
            null,
            0,
            0,
            0,
            0,
            null,
            null,
            0
        ]
    },
    "files": [
        {
            "name": "app/calculator.ts",
            "totals": {
                "files": 0,
                "lines": 10,
                "hits": 8,
                "misses": 1,
                "partials": 1,
                "coverage": 80,
                "branches": 1,
                "methods": 0,
                "messages": 0,
                "sessions": 0,
                "complexity": 0,
                "complexity_total": 0,
                "complexity_ratio": 0,
                "diff": 0
            }
        }
    ],
    "commit_file_url": "https://app.codecov.io/github/codecov/example-typescript/commit/69441683464ac4ba859654416c77fa0d4e732882/tree/"
}
```

The only file lacks the `line_coverage` property promised in the docs, without which Reviewable won't be able to do much.

Are you able to get this API to work for your repo?  If so, could you please share the request and response either here or via email to support@reviewable.io?  Thanks!

**Comment by outofforest** [comment]
Hi @pkaminski 

Here are my results:

```
curl 'https://api.codecov.io/api/v2/github/outofforest/repos/volume/totals/?sha=5eb25349aad3a03b71ef6dcd37a689b1905b16d3' -H 'accept: application/json' | jq
{
  "totals": {
    "files": 4,
    "lines": 118,
    "hits": 99,
    "misses": 15,
    "partials": 4,
    "coverage": 83.9,
    "branches": 0,
    "methods": 0,
    "messages": 0,
    "sessions": 1,
    "complexity": 0.0,
    "complexity_total": 0.0,
    "complexity_ratio": 0,
    "diff": [
      0,
      0,
      0,
      0,
      0,
      null,
      0,
      0,
      0,
      0,
      null,
      null,
      0
    ]
  },
  "files": [
    {
      "name": "api.go",
      "totals": {
        "files": 0,
        "lines": 37,
        "hits": 27,
        "misses": 8,
        "partials": 2,
        "coverage": 72.97,
        "branches": 0,
        "methods": 0,
        "messages": 0,
        "sessions": 0,
        "complexity": 0.0,
        "complexity_total": 0.0,
        "complexity_ratio": 0,
        "diff": 0
      }
    },
    {
      "name": "reduce.go",
      "totals": {
        "files": 0,
        "lines": 44,
        "hits": 42,
        "misses": 2,
        "partials": 0,
        "coverage": 95.45,
        "branches": 0,
        "methods": 0,
        "messages": 0,
        "sessions": 0,
        "complexity": 0.0,
        "complexity_total": 0.0,
        "complexity_ratio": 0,
        "diff": 0
      }
    },
    {
      "name": "lib/libnet/net.go",
      "totals": {
        "files": 0,
        "lines": 4,
        "hits": 4,
        "misses": 0,
        "partials": 0,
        "coverage": 100.0,
        "branches": 0,
        "methods": 0,
        "messages": 0,
        "sessions": 0,
        "complexity": 0.0,
        "complexity_total": 0.0,
        "complexity_ratio": 0,
        "diff": 0
      }
    },
    {
      "name": "lib/libhttp/http.go",
      "totals": {
        "files": 0,
        "lines": 33,
        "hits": 26,
        "misses": 5,
        "partials": 2,
        "coverage": 78.79,
        "branches": 0,
        "methods": 0,
        "messages": 0,
        "sessions": 0,
        "complexity": 0.0,
        "complexity_total": 0.0,
        "complexity_ratio": 0,
        "diff": 0
      }
    }
  ],
  "commit_file_url": "https://app.codecov.io/github/outofforest/volume/commit/5eb25349aad3a03b71ef6dcd37a689b1905b16d3/tree/"
}
```

Indeed, `line_coverage` is missing. My bet is that to optimize the traffic they return this information only if some additional parameter is added to the URL. Maybe they forgot to document it.

**Comment by pkaminski** [comment]
Got it -- we need to use a [different API endpoint](https://docs.codecov.com/reference/repos_commits_retrieve).  Amusingly, they also changed the meaning of the per-line values so that whereas `0` used to mean "no coverage" it now means "full coverage".  Anyway, I'm traveling at the moment, but should be able to add support for the v2 API next week.  Thanks for your patience!

```
$ curl 'https://api.codecov.io/api/v2/github/codecov/repos/example-typescript/commits/69441683464ac4ba859654416c77fa0d4e732882' -H 'accept: application/json'

{
    "commitid": "69441683464ac4ba859654416c77fa0d4e732882",
    "message": "chore: update CI",
    "timestamp": "2022-06-10T03:33:42Z",
    "ci_passed": true,
    "author": {
        "service": "github",
        "username": "thomasrockhu-codecov",
        "name": "Tom Hu"
    },
    "branch": "master",
    "totals": {
        "files": 1,
        "lines": 10,
        "hits": 8,
        "misses": 1,
        "partials": 1,
        "coverage": 80,
        "branches": 1,
        "methods": 0,
        "sessions": 3,
        "complexity": 0,
        "complexity_total": 0,
        "complexity_ratio": 0,
        "diff": 0
    },
    "state": "complete",
    "parent": "1975e3b2b0f4baaf0e71218a6710d26f4a708b42",
    "report": {
        "totals": {
            "files": 1,
            "lines": 10,
            "hits": 8,
            "misses": 1,
            "partials": 1,
            "coverage": 80,
            "branches": 1,
            "methods": 0,
            "messages": 0,
            "sessions": 3,
            "complexity": 0,
            "complexity_total": 0,
            "complexity_ratio": 0,
            "diff": [
                0,
                0,
                0,
                0,
                0,
                null,
                0,
                0,
                0,
                0,
                null,
                null,
                0
            ]
        },
        "files": [
            {
                "name": "app/calculator.ts",
                "totals": {
                    "files": 0,
                    "lines": 10,
                    "hits": 8,
                    "misses": 1,
                    "partials": 1,
                    "coverage": 80,
                    "branches": 1,
                    "methods": 0,
                    "messages": 0,
                    "sessions": 0,
                    "complexity": 0,
                    "complexity_total": 0,
                    "complexity_ratio": 0,
                    "diff": 0
                },
                "line_coverage": [
                    [
                        1,
                        0
                    ],
                    [
                        2,
                        0
                    ],
                    [
                        5,
                        0
                    ],
                    [
                        6,
                        0
                    ],
                    [
                        9,
                        0
                    ],
                    [
                        10,
                        0
                    ],
                    [
                        13,
                        0
                    ],
                    [
                        14,
                        2
                    ],
                    [
                        15,
                        1
                    ],
                    [
                        17,
                        0
                    ]
                ]
            }
        ]
    }
}
```

**Comment by outofforest** [comment]
Thanks in advance!

**Comment by pkaminski** [comment]
I deployed code that suggests Codecov API v2 URLs by default and understands the new schema.  However, there's a CORS bug in their API server that will prevent requests emitted from the browser from succeeding.  I raised an [issue](https://community.codecov.com/t/cors-error-when-making-api-v2-requests/4415) about it but don't know if/when it'll be addressed.  In the meantime, Codecov v2 should work for private repos but not for public ones.  Hopefully that's good enough for you -- otherwise feel free to help make noise about their bug!
